<p>This rule detects usage of class.getClassLoader invocation</p>
<h2>Noncompliant Code Example</h2>
<pre id="compliantContent">

    // Koupleless runtime engage a multi app pattern, each app is binded with a unique class loader.
    // Therefore classes in app0 can't be founded by the class loader of app1, or else error would be thrown.
    // To ensure the biz logic correctness, each classLoader assignment should be carefully checked.

    SomeClass.class.getClassLoader()
</pre>
<h2>Compliant Solution</h2>
<pre id="compliantSolution">

    // To fix this issue, we should find a way to use a more proper class loader to perform the further logic.
    Thread.currentThread().getContextClassLoader() // this is usually correct
</pre>
